lib/src/patterns/observer/Notification.dart

```diff
@@ -19,80 +19,73 @@ part of puremvc;
  *
  * See [IView], [IObserver], [Notification]
  */
-class Notification implements INotification
-{
-    /**
-     * Constructor.
-     *
-     * -  Param [name] - name of the [INotification].
-     * -  Param [body] - the [INotification] body. (optional)
-     * -  Param [type] - the type of the [INotification] (optional)
-     */
-    Notification( String this.name, [dynamic this.body, String this.type] ){}
+class Notification implements INotification {
+  /**
+   * Constructor.
+   *
+   * -  Param [_name] - name of the [INotification].
+   * -  Param [_body] - the [INotification] body. (optional)
+   * -  Param [_type] - the type of the [INotification] (optional)
+   */
+  Notification(this._name, [dynamic this._body, this._type]) {}
 
-    /**
-     * Get the [name] of the [INotification].
-     *
-     * -  Returns [String] - the name of the [INotification].
-     */
-    String getName()
-    {
-        return name;
-    }
+  /**
+   * Get the [name] of the [INotification].
+   *
+   * -  Returns [String] - the name of the [INotification].
+   */
+  String getName() {
+    return _name;
+  }
 
-    /**
-     * Set the [body] of the [INotification].
-     *
-     * -  Param [body] - the body of the [INotification].
-     */
-    void setBody( dynamic bodyObject )
-    {
-        body = bodyObject;
-    }
+  /**
+   * Set the [body] of the [INotification].
+   *
+   * -  Param [body] - the body of the [INotification].
+   */
+  void setBody(dynamic bodyObject) {
+    _body = bodyObject;
+  }
 
-    /**
-     * Get the [body] of the [INotification].
-     *
-     * -  Returns [Dynamic] - the body of the [INotification].
-     */
-    dynamic getBody()
-    {
-        return body;
-    }
+  /**
+   * Get the [body] of the [INotification].
+   *
+   * -  Returns [Dynamic] - the body of the [INotification].
+   */
+  dynamic getBody() {
+    return _body;
+  }
 
-    /**
-     * Set the [type] of the [INotification].
-     *
-     * -  Param [type] - the type of the [INotification].
-     */
-    void setType( String noteType )
-    {
-        type = noteType;
-    }
+  /**
+   * Set the [type] of the [INotification].
+   *
+   * -  Param [type] - the type of the [INotification].
+   */
+  void setType(String? noteType) {
+    _type = noteType;
+  }
 
-    /**
-     * Get the [type] of the [INotification].
-     *
-     * -  Returns [String] - the type of the [INotification].
-     */
-    String getType()
-    {
-        return type;
-    }
+  /**
+   * Get the [type] of the [INotification].
+   *
+   * -  Returns [String] - the type of the [INotification].
+   */
+  String? getType() {
+    return _type;
+  }
 
-    /**
-     * This [INotifications]'s [name]
-     */
-    String name;
+  /**
+   * This [INotifications]'s [name]
+   */
+  final String _name;
```

Comment:
why are you adding final to _name but not _type? I think dropping it from both is what's needed.